app/helpers/public_document_routes_helper.rb

```diff
@@ -40,7 +40,7 @@ def preview_document_url(edition, options = {})
     if edition.rendering_app == Whitehall::RenderingApp::GOVERNMENT_FRONTEND
       options[:host] = URI(Plek.new.external_url_for("draft-origin")).host
     else
-      options[:preview] = edition.latest_edition.id
+      options[:preview] = edition.document.latest_edition.id
```

Comment:
Not sure if the edition is already in memory here, but if not then 

```
edition.document.latest_edition_id
```

would be quicker

Reply:
Good spot – thanks. I've just changed that.

And this will hopefully go away altogether once frontend rendering has been removed from Whitehall ✨